packages/jest-jasmine2/src/setup-jest-globals.js

```diff
@@ -48,67 +48,17 @@ const addSuppressedErrors = result => {
   }
 };
 
-function addAssertionErrors(result) {
-  const {
-    assertionCalls,
-    expectedAssertionsNumber,
-    isExpectingAssertions,
-  } = getState();
-  setState({
-    assertionCalls: 0,
-    expectedAssertionsNumber: null,
-  });
-  if (
-    typeof expectedAssertionsNumber === 'number' &&
-    assertionCalls !== expectedAssertionsNumber
-  ) {
-    const expected = EXPECTED_COLOR(
-      pluralize('assertion', expectedAssertionsNumber),
-    );
-    const message = new Error(
-      matcherHint('.assertions', '', String(expectedAssertionsNumber), {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but only received ` +
-        RECEIVED_COLOR(pluralize('assertion call', assertionCalls || 0)) +
-        '.',
-    ).stack;
-    result.status = 'failed';
-    result.failedExpectations.push({
-      actual: assertionCalls,
-      expected: expectedAssertionsNumber,
-      message,
-      passed: false,
-    });
-  }
-  if (isExpectingAssertions && assertionCalls === 0) {
-    const expected = EXPECTED_COLOR('at least one assertion');
-    const received = RECEIVED_COLOR('received none');
-    const message = new Error(
-      matcherHint('.hasAssertions', '', '', {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but ${received}.`,
-    ).stack;
-    result.status = 'failed';
-    result.failedExpectations.push({
-      actual: 'none',
-      expected: 'at least one',
-      message,
-      passed: false,
-    });
-  }
-}
+const addAssertionErrors = () => {
+  extractExpectedAssertionsErrors();
```

Comment:
this is only the first half of what needs to be done.
This function will return you a list of errors (but won't throw anything).
After that you need to take these errors and put them into Jasmine test result (see `result.failedExpectations.push(...)` in the old code)